### PR TITLE
 Small usability improvements

### PR DESCRIPTION
**Pull request by dnknth**
Add fat Jar to gradle build, add arguments to `UpdateWithAdditionalOptions`

### CODE REVIEW

**Comment by signed** [comment]
@dnknth  thank you for the PR. I'm open to add the command line interface if it helps you. But its not my preferred way to use this project. I usually make changes in  `UpdateWithAdditionalOptions.java` and run it from the IDE. Therefore two asks from my side:

1. Please move your changes into a new class like `CommandLineRunner` or `CommandLineInterface` and keep `UpdateWithAdditionalOptions.java` as is
3. Add some documentation to the readme  so other people can learn that there is a fat jar with a command line interface.

File: src/main/java/jb/UpdateWithAdditionalOptions.java [hunk 1]
```diff
@@ -3,15 +3,82 @@
...
+    }
+
+    @Parameter(description = "directory...")
+    private List<String> parameters = new ArrayList<>();
+
+    @Parameter(names = { "-h", "--help" }, help = true, description = "Show this help message")
+    private boolean help = false;
+
+    @Parameter(names = { "-n", "--dryrun" }, description = "Do not change files, just report what would happen")
+    private boolean dryRun = false;
```
Comment (signed) [inline src/main/java/jb/UpdateWithAdditionalOptions.java:35]:
```suggestion
    private boolean dryRun = true;
```
Just a proposal, I made some bad experiences with command line tools that where destructive by default.

File: src/main/java/jb/UpdateWithAdditionalOptions.java [hunk 2]
```diff
@@ -3,15 +3,82 @@
...
+    }
+
+    public static void main(String[] argv) throws IOException {
+        UpdateWithAdditionalOptions main = new UpdateWithAdditionalOptions();
+        JCommander jCommander = JCommander.newBuilder()
+            .addObject(main)
+            .build();
+
+        jCommander.parse(argv);
+        if (main.help) {
```
Comment (signed) [inline src/main/java/jb/UpdateWithAdditionalOptions.java:78]: Print the help in case no arguments are passed?

File: build.gradle.kts [hunk 3]
```diff
@@ -21,6 +21,7 @@ dependencies {
     implementation(group = "com.github.javaparser", name = "javaparser-core", version = "3.16.2")
     implementation(group = "org.hamcrest", name = "hamcrest-all", version = "1.3")
     implementation(group = "commons-io", name = "commons-io", version = "2.5")
+    implementation("com.beust:jcommander:1.78")
```
Comment (Bukama) [inline build.gradle.kts:24]: Please align the coordinates to the style we used before

**Review by Bukama — COMMENTED** [review]
Just a small note.

**Comment by signed** [comment]
Thank you for the updates. 
If you add a small update to the readme this should be good to merge.

Add a new heading like `Running the update from the command line, document the command how to build the fat jar and link to `src/main/java/jb/CommandLineRunner.java` that people can see what parameters are available.

**Comment by signed** [comment]
@dnknth  thank you for you contribution.

**Pull request merged**